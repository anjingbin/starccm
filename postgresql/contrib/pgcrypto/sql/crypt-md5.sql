--
-- crypt() and gen_salt(): md5
--

select crypt('', '$1$Szzz0yzz');

select crypt('foox', '$1$Szzz0yzz');

create table ctest (data text, res text, salt text);
insert into ctest values ('password', '', '');

update ctest set salt = gen_salt('md5');
update ctest set res = crypt(data, salt);
select res = crypt(data, res) as "worked" from ctest;

drop table ctest;

