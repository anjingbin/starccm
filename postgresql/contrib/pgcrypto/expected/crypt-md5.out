--
-- crypt() and gen_salt(): md5
--
select crypt('', '$1$Szzz0yzz');
               crypt                
------------------------------------
 $1$Szzz0yzz$To38XrR3BsbXQW2ZpfKjF1
(1 row)

select crypt('foox', '$1$Szzz0yzz');
               crypt                
------------------------------------
 $1$Szzz0yzz$IYL49cd3t9bllsA7Jmz1M1
(1 row)

create table ctest (data text, res text, salt text);
insert into ctest values ('password', '', '');
update ctest set salt = gen_salt('md5');
update ctest set res = crypt(data, salt);
select res = crypt(data, res) as "worked" from ctest;
 worked 
--------
 t
(1 row)

drop table ctest;
